zeppelin-server/src/main/java/com/nflabs/zeppelin/rest/InterpreterRestApi.java

```diff
@@ -87,6 +87,23 @@ public Response newSettings(String message) throws InterpreterException, IOExcep
     return new JsonResponse(Status.CREATED, "").build();
   }
 
+  @PUT
+  @Path("setting/{settingId}")
+  public Response updateSetting(String message, @PathParam("settingId") String settingId) {
+    logger.info("Update interpreterSetting {}", settingId);
+    try {
+      Properties p = gson.fromJson(message, Properties.class);
+      interpreterFactory.setPropertyAndRestart(settingId, p);
+    } catch (InterpreterException e) {
```

Comment:
Maybe it is good to start to handle different error, for exemple the `JsonSyntaxException`, imagine in my request i send  ``Yo died``, then i will get a 500, maybe we can catch this and return a proper http code like `400 Bad Request`